core/src/main/java/org/elasticsearch/common/cli/Terminal.java

```diff
@@ -144,27 +103,21 @@ public String readText(String text, Object... args) {
         public char[] readSecret(String text, Object... args) {
             return console.readPassword(text, args);
         }
-
-        @Override
-        public void printStackTrace(Throwable t) {
-            t.printStackTrace(console.writer());
-        }
     }
 
-    @SuppressForbidden(reason = "System#out")
     private static class SystemTerminal extends Terminal {
 
-        private final PrintWriter printWriter = new PrintWriter(System.out);
-
         @Override
+        @SuppressForbidden(reason = "System#out")
         public void doPrint(String msg) {
             System.out.print(msg);
+            System.out.flush();
         }
 
         @Override
         public String readText(String text, Object... args) {
```

Comment:
Good call, I had not noticed that. I've removed the format args from those methods.